zcash_client_sqlite/src/wallet/init/migrations/v_transactions_transparent_history.rs

```diff
@@ -34,6 +34,8 @@ impl RusqliteMigration for Migration {
     type Error = WalletMigrationError;
 
     fn up(&self, transaction: &rusqlite::Transaction) -> Result<(), Self::Error> {
+        // FIXME: This will *not* include a transaction by virtue of it involving a utxo
+        // that is in the utxos table with unknown height. Is that correct?
```

Comment:
I think this comment is incorrect? We `LEFT JOIN blocks ON blocks.height = notes.block`, so when `notes.block = -1` there won't be a matching entry in `blocks`, and we will include the transaction but fill with `NULL`.

However, I think the view is still then incorrect: users of the view will see `mined_height` set to both `NULL` and `-1` meaning "unmined (from the wallet's view)" instead of just `NULL`. I think the fix is a migration that changes this view to use `blocks.height AS mined_height`, which will be NULL in both cases (post-`LEFT JOIN`).

Aside: existing migrations are the wrong place to add comments like this, because these migrations are fixed and won't semantically change. If you want to annotate the current database state, do so with the pinned database state in `zcash_client_sqlite/src/wallet/init.rs`.

Reply:
Yes, I had missed that it was a LEFT JOIN.

I was intending all of these comments to be removed before merging (and in fact they will be replaced by #1402).